### DISCUSSION TITLE
which telemetry setting shall I use when sending error telemetry from extension

### DISCUSSION THREAD

**Discussion by nancechen** [microsoft/vscode-discussions#369] [Extension Development QnA] [answered]
Hey, I'm writing a vscode extension and want to send telemetry on usage data and errors. 
From the doc [here](https://code.visualstudio.com/api/extension-guides/telemetry), it says use the isTelemetryEnabled api
![image](https://user-images.githubusercontent.com/121999432/210707121-57341aa9-1a86-4aba-998f-8c422f0f0302.png). My telemetryClient check isTelemetryEnabled and then decide whether or not to send telemetry. From testing, I found isTelemetryEnabled only returns true when telemetry level is set to 'all'. When user sets the telemetry setting to error, my understanding is that I can send error telemetry but I  have to get this value from telemetry.telemetryLevel. However, from the doc, it seems like the telemetry setting value is not reliable. 
![image](https://user-images.githubusercontent.com/121999432/210707786-e53ff271-e766-4f63-b1bc-adc96227546b.png)
My question is: when I send error telemetry, which telemetry setting I should be checking?

**Comment by burkeholland** [comment]
I think @isidorn might know this one

**Comment by isidorn** [comment, marked as answer]
@nancechen when you send error telemetry use the `telemetry.telemetryLevel` setting
@lramos15 can correct me if wrong.
Thanks

**Reply by nancechen** [reply]
thanks, I was planning to use this setting, but found out onDidChangeTelemetryEnabled api does not get triggered when telemetry level changes among error, crash and off

**Reply by lramos15** [reply]
Sadly this is currently a hole in our telemetry API surface. It will be fixed shortly by the finalization of https://github.com/microsoft/vscode/issues/160090, but until then extensions are limited to sending error telemetry only when `isTelemetryEnabled` is true which is only when a user has consented to both usage and error telemetry. This is because a user can launch vscode with specific flags such as `--disable-telemetry` and in that case the setting is incorrect.
